Godot Skeleton v1.1.0

brettchalupa opened the pull request:
I'm working on the next release of Godot Skeleton--v1.1.0. The PR will be in draft mode while I'm working on it, but I figure it might be fun to follow along. Goals are mostly to improve what's already there. No major overhauls. See commits for more details.

brettchalupa committed:
chore: use compatibility renderer

it has the most compatibility, so I think it's a sensible default. it's
easy enough to change as needed

brettchalupa committed:
chore(docs): mention web build issues on macOS

hopefully this gets fixes/improved soon by either the browser vendors or
godot!

brettchalupa committed:
feat(sound): Sound.play_sfx & example usage

this adds a global singleton func for easily playing sfx that respects
the player's setting. included is a select sfx that's called when
pausing the game to test and verify its usage

jeff-kwak commented on sound.gd:
```diff
@@ -0,0 +1,16 @@
+extends Node
```
just curious; why not use `class_name Sound` (or similar) here.

brettchalupa replied:
Good feedback, I'll add that! That'll make it more clear. Godot automatically makes it work when setting up the autoload singleton, but I like the idea of it being named.

brettchalupa replied:
I dropped this in, `class_name Sound extends Node` based on your feedback, and it returns this error:

> res://sound.gd:1 - Parse Error: Class "Sound" hides an autoload singleton.

Kind of interesting that it doesn't let ya' do it for autoload singletons. Good learning! :smile:

brettchalupa committed:
feat: display ver metadata from export.cfg in main menu

this lets players and developers know which version of a game they're
running. this can be helpful with bug reports and just in general having
confidence the proper version has been uploaded

brettchalupa committed:
feat(sound): add Sound.play_music

this respects the player's setting, much like `Sound.play_sfx`

brettchalupa committed:
chore: consistent file naming

uses snake_case and documents this in the README

brettchalupa committed:
chore: add type signatures to GDScript code

this is my preference due to the clarity about what's being worked with
and the better editor integration

brettchalupa committed:
feat(debug): group-driven debug UI

toggleable with the 0 keyboard key; really useful for showing special UI
when working on a game. hidden by default. easily adjustable

brettchalupa commented:
I'm feeling pretty good about these changes. Going to merge this into `main` so its the new default version. Will work on v1.2 after I finish my next game, where I'm sure I'll learn more and merge it back into this.
